add example for setting up grafana dashboard (#1243)
kyverno/website: content/en/docs/monitoring/bonus-grafana-dashboard/index.md

![Dashboard example 1](dashboard-example-1.png)

![Dashboard example 2](dashboard-example-2.png)

> ### Example: Setup Grafana with Kyverno and a Kind cluster

Reviewer: ```suggestion
### Tutorial
```

```sh
kind create cluster
```

> Add helm repositories

Reviewer: ```suggestion
Add Helm repositories.
```

```sh
helm repo add prometheus-community https://prometheus-community.github.io/helm-charts 
helm repo add kyverno https://kyverno.github.io/kyverno/ 
```

> Update repos

Reviewer: ```suggestion
Update Helm repositories.
```

```sh
helm repo update    
```

> Install Kyverno and monitoring

Reviewer: ```suggestion
Install Kyverno and the kube-prometheus-stack Helm chart.
```

```sh
helm install kyverno kyverno/kyverno --namespace kyverno --create-namespace 
helm install monitoring prometheus-community/kube-prometheus-stack -n monitoring --create-namespace

> k -n monitoring get po -l "release"

Reviewer: ```suggestion
kubectl -n monitoring get po -l "release"
```

```

> Create the service monitor

Reviewer: ```suggestion
Create the ServiceMonitor resource.
```

> `vi sm.yaml`

Reviewer: ```suggestion
```

> Add the following service monitor yaml

Reviewer: ```suggestion
Add the following contents to a file named `service-monitor.yaml`.
```

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    app.kubernetes.io/instance: monitoring
    chart: kube-prometheus-stack-51.2.0
    heritage: Helm
    release: monitoring
  name: service-monitor-kyverno-service
  namespace: monitoring
spec:
  endpoints:
  - interval: 60s
    path: /metrics
    scheme: http
    targetPort: 8000
    tlsConfig:
      insecureSkipVerify: true
  namespaceSelector:
    matchNames:
    - kyverno
  selector:
    matchLabels:
      app.kubernetes.io/instance: kyverno
```

> Add right labels

Reviewer: ```suggestion
Add the appropriate labels.
```

```sh

> k label ns kyverno app.kubernetes.io/instance=kyverno                                                               
> k label ns kyverno app.kubernetes.io/name=kyverno

Reviewer: ```suggestion
kubectl label ns kyverno app.kubernetes.io/instance=kyverno                                                               
kubectl label ns kyverno app.kubernetes.io/name=kyverno
```

```

> Apply service-monitor.yaml

Reviewer: ```suggestion
Create the ServiceMonitor.
```

```sh

> k apply -f sm.yaml

Reviewer: ```suggestion
kubectl apply -f service-monitor.yaml
```

```

> Restart deployments and sts in monitoring ns

Reviewer: ```suggestion
Restart Deployments and StatefulSets in the `monitoring` Namespace.
```

```sh

> k rollout restart deploy,sts -n monitoring

Reviewer: ```suggestion
kubectl rollout restart deploy,sts -n monitoring
```

```

> Check services in monitoring ns

Reviewer: ```suggestion
Check services in the `monitoring` Namespace.
```

```sh

> k get svc -n monitoring             

Reviewer: ```suggestion
kubectl get svc -n monitoring             
```

                                                                                                            
NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
monitoring-kube-prometheus-prometheus     ClusterIP   10.96.238.189   <none>        9090/TCP,8080/TCP            4h16m
```

> Port forward the svc/monitoring-kube-prometheus-prometheus to a local port

Reviewer: ```suggestion
Port forward the `monitoring-kube-prometheus-prometheus` Service to a local port.
```

```sh

> sudo kubectl port-forward svc/monitoring-kube-prometheus-prometheus 81:9090 -n monitoring                                                       

Reviewer: ```suggestion
kubectl port-forward svc/monitoring-kube-prometheus-prometheus 81:9090 -n monitoring                                                       
```

Forwarding from 127.0.0.1:81 -> 9090
Forwarding from [::1]:81 -> 9090
```

> Similarly port forward the svc/monitoring-grafana to another local port

Reviewer: ```suggestion
Similarly, port forward the `monitoring-grafana` Service to another local port.
```

```sh

> k get svc -n monitoring               

Reviewer: ```suggestion
kubectl get svc -n monitoring               
```

                                                                             
NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
monitoring-grafana                        ClusterIP   10.96.188.20    <none>        80/TCP                       4h19m

> sudo kubectl port-forward svc/monitoring-grafana -n monitoring 82:80    

Reviewer: ```suggestion
kubectl port-forward svc/monitoring-grafana -n monitoring 82:3000
```

Forwarding from 127.0.0.1:82 -> 3000
Forwarding from [::1]:82 -> 3000
```

> Go to Prometheus on port 81 (in this case) and check status -> Targets -> Filter for kyverno (to see if metrics are getting scraped)

Reviewer: ```suggestion
Go to Prometheus on port 9090 and check status -> Targets -> Filter for kyverno (to see if metrics are getting scraped)
```

> Go to Grafana on port 82 (in this case) -> Dashboards -> New ->  import -> Upload file that you get from running the below command -> Data type = Prometheus -> import

Reviewer: ```suggestion
Go to Grafana on port 3000 -> Dashboards -> New ->  import -> Upload file that you get from running the below command -> Data type = Prometheus -> import
```
